hack/jenkins/job-configs/kubernetes-upgrades.yaml

```diff
@@ -109,72 +117,120 @@
+            job-env: |
+                export E2E_OPT="--check_version_skew=false"
+                export E2E_UP="false"
+                export E2E_DOWN="false"
+                export GINKGO_TEST_ARGS="--ginkgo.focus=Kubectl"
         - 'kubernetes-upgrade-{provider}-{version-old}-{version-new}-{step}':
             step: 'step3-upgrade-master'
-            branch: '{branch-new}'
             description: 'Upgrade the master from {version-old} to {version-new}.'
+            job-env: |
+                export E2E_OPT="--check_version_skew=false"
+                export E2E_UP="false"
+                export E2E_DOWN="false"
+                export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:Upgrade\].*upgrade-master"
         - 'kubernetes-upgrade-{provider}-{version-old}-{version-new}-{step}':
             step: 'step4-e2e-old'
-            branch: '{branch-old}'
+            runner: '{runner-old}'
             description: 'Run {version-old} e2e tests against the cluster with master at {version-new} and nodes still at {version-old}.'
+            job-env: |
+                export E2E_OPT="--check_version_skew=false"
+                export E2E_UP="false"
+                export E2E_DOWN="false"
         - 'kubernetes-upgrade-{provider}-{version-old}-{version-new}-{step}':
             step: 'step5-upgrade-cluster'
-            branch: '{branch-new}'
             description: 'Upgrade the nodes from {version-old} to {version-new}.'
+            job-env: |
+                export E2E_OPT="--check_version_skew=false"
+                export E2E_UP="false"
+                export E2E_DOWN="false"
+                export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:Upgrade\].*upgrade-cluster"
         - 'kubernetes-upgrade-{provider}-{version-old}-{version-new}-{step}':
             step: 'step6-e2e-old'
-            branch: '{branch-old}'
+            runner: '{runner-old}'
             description: 'Run {version-old} e2e tests against the cluster with master and nodes at {version-new}.'
+            job-env: |
+                export E2E_OPT="--check_version_skew=false"
+                export E2E_UP="false"
+                export E2E_DOWN="false"
         - 'kubernetes-upgrade-{provider}-{version-old}-{version-new}-{step}':
             step: 'step7-e2e-new'
-            branch: '{branch-new}'
             description: 'Run {version-new} e2e tests against the cluster with master and nodes at {version-new}.'
+            job-env: |
+                # TODO(15011): these really shouldn't be (very) version skewed, but
+                # because we have to get ci/latest again, it could get slightly out of
+                # whack.
+                export E2E_OPT="--check_version_skew=false"
+                export E2E_UP="false"
 
 - project:
     name: 'upgrade-gke'
+    provider-env: |
+        {gke-provider-env}
+        export JENKINS_FORCE_GET_TARS="y"
+        export FAIL_ON_GCP_RESOURCE_LEAK="false"
+    project-env: ''
```

Comment:
This is confusing to me; I assume it's for defaulting, but maybe just have no default, and stick an empty `project-env` into the pre-v1.2 projects?